app/src/components/market/market_buy.tsx

```diff
@@ -152,7 +153,29 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
     }
   }
 
-  const disabled = (status !== Status.Ready && status !== Status.Error) || cost.isZero()
+  const calculateCollateralBalance = useMemo(
```

Comment:
This seems to work, but the problem is that it kind of duplicates the logic inside `BalanceToken`. Maybe `BalanceToken` should receive the user's balance instead of computing it? (I like this idea because it would make it more pure).

If we do that, then we should also do it in the sell and fund pages. But that makes sense: this check should be there too.

Reply:
Just to be clearer: if we make this, `BalanceToken` should receive everything it needs. It shouldn't have any `useEffect` or `useAsyncDerivedValue`.

And btw I'm not sure if we need `useAsyncDerivedValue` for this, since it delays the computation unnecessarily. (I blame me for not indicating this in the name of the hook, but `useAsyncAndDelayedDerivedValue` doesn't sound that good).

Reply:
I create a hook to get the balance of some collateral. `BalanceToken` component is more pure right now, only receive balance and collateral information and doesn't have any `useEffect` or `useAsyncDerivedValue`.